### ISSUE TITLE
Repeat not working when task is checked in Daily Note

### ISSUE THREAD

**Issue by raph90** [NotePlan/plugins#532]
**Describe the bug**
If I have a note, called 'Home Maintenance' for example, and I create a repeating todo there, and schedule it for today, if I now navigate to the Daily Note for the day it's scheduled for (today), and check it, then the note will not automatically generate a new repeat. I think this is because the page that it's originally on (Home Maintenance) hasn't actually been saved...

Is there anyway around this?

Skipping the following section because I'm _not_ quite sure if this is a bug or a feature request. 

Thanks! (Great extension btw).

**Comment by jgclark** [comment]
I'm just double checking that you're using the @repeat(...) syntax, not the built-in one?
And that you have a Repeats plugin trigger on the note in question?
If so, then this is expected behaviour, as the trigger only fires when you have the note open in a NP window and change it.
I've asked @EduardMe a number of times for an "onSave" trigger for exactly this sort of reason.

**Comment by raph90** [comment]
Hi @jgclark, yep this is what I'm doing:
<img width="434" alt="Screenshot 2024-02-24 at 22 49 04" src="https://github.com/NotePlan/plugins/assets/30865601/a3fc3ecb-f4b6-433f-bc30-d0b769074b31">

Then on the today page I get:
<img width="449" alt="Screenshot 2024-02-24 at 22 49 23" src="https://github.com/NotePlan/plugins/assets/30865601/0919c8b3-ee72-4fd3-bf00-6b207f7fc0e5">

If I click done on that todo, I get this:
<img width="441" alt="Screenshot 2024-02-24 at 22 49 40" src="https://github.com/NotePlan/plugins/assets/30865601/51547d46-69e8-42ef-acd2-8bc2f7b7f449">

But if I then go back to the self maintenance page, I see this:
<img width="404" alt="Screenshot 2024-02-24 at 22 50 04" src="https://github.com/NotePlan/plugins/assets/30865601/91c2079c-4591-45b7-a7c4-36ee1f9d1acb">

So basically it doesn't update. Does my syntax all look correct?

**Comment by jgclark** [comment]
Thanks. You confirm that this is due to a limitation with the current app triggers.

**Comment by cwhittl** [comment]
Having the same issue, is this something being worked on (the triggers limitation)?

**Comment by jgclark** [comment]
> Having the same issue, is this something being worked on (the triggers limitation)?

I can't change the way triggers work: only the app developer can do that ...

... However ... 

I've now added a **/Generate @repeats in recent notes** command to the Tidy plugin for you @raph90. This generates any needed new @repeat() lines in all recently-changed notes.
It means you don't have to add the special trigger on every relevant note: instead you can include this command in your Daily Note template, or run it manually every day or two.

More details at https://github.com/NotePlan/plugins/blob/main/np.Tidy/README.md, including how you can contribute to my coffee fund if you'd like to ;-)
